p11-kit/list-objects.c

```diff
@@ -191,16 +189,16 @@ get_object_uri (P11KitIter *iter,
 		goto cleanup;
 	}
 
-	if (label != NULL) {
-	        ret = p11_kit_uri_set_attribute (uri, &attr_label);
+	if (attr_label->ulValueLen != CK_UNAVAILABLE_INFORMATION) {
+	        ret = p11_kit_uri_set_attribute (uri, attr_label);
 	        if (ret != P11_KIT_URI_OK) {
 		        p11_message (_("failed to set attribute for URI: %s"), p11_kit_uri_message (ret));
 		        goto cleanup;
 	        }
 	}
 
-	if (id != NULL) {
-		ret = p11_kit_uri_set_attribute (uri, &attr_id);
+	if (attr_id->ulValueLen != CK_UNAVAILABLE_INFORMATION) {
```

Comment:
This should be fine as well as the `pValue` should always be allocated in case `ulValueLen` is set